app/javascript/crayons/MentionAutocompleteTextArea/MentionAutocompleteTextArea.jsx

```diff
@@ -54,284 +73,311 @@ const replaceTextArea = ({
+      inputProps.defaultValue ? inputProps.defaultValue : '',
+    );
+    const [searchTerm, setSearchTerm] = useState('');
+    const [cachedSearches, setCachedSearches] = useState({});
+    const [dropdownPositionPoints, setDropdownPositionPoints] = useState({
+      x: 0,
+      y: 0,
+    });
+    const [selectionInsertIndex, setSelectionInsertIndex] = useState(0);
+    const [users, setUsers] = useState([]);
+    const [cursorPosition, setCursorPosition] = useState(null);
+    const [ariaHelperText, setAriaHelperText] = useState('');
 
-    fetchSuggestions(searchTerm).then(({ result: fetchedUsers }) => {
-      const resultLength = Math.min(fetchedUsers.length, MAX_RESULTS_DISPLAYED);
+    const isSmallScreen = useMediaQuery(`(max-width: ${BREAKPOINTS.Small}px)`);
 
-      const results = fetchedUsers.slice(0, resultLength);
+    const plainTextAreaRef = useRef(null);
+    const comboboxRef = useRef(null);
+    const popoverRef = useRef(null);
+    const containerRef = useRef(null);
 
-      setCachedSearches({
-        ...cachedSearches,
-        [searchTerm]: results,
-      });
+    const { setTextArea, setAdditionalElements } = useTextAreaAutoResize();
 
-      setUsers(results);
+    const { onChange, id: inputId, ...autocompleteInputProps } = inputProps;
 
-      // Let screen reader users know a list has populated
-      if (!ariaHelperText && fetchedUsers.length > 0) {
-        setAriaHelperText(`Mention user, ${fetchedUsers.length} results found`);
+    useLayoutEffect(() => {
+      if (autoResize && comboboxRef.current && plainTextAreaRef.current) {
+        setTextArea(plainTextAreaRef.current);
+        setAdditionalElements([comboboxRef.current, containerRef.current]);
       }
-    });
-  }, [searchTerm, fetchSuggestions, cachedSearches, ariaHelperText]);
-
-  useLayoutEffect(() => {
-    const popover = popoverRef.current;
-    if (!popover) {
-      return;
-    }
+    }, [autoResize, setTextArea, setAdditionalElements]);
 
-    const closeOnClickOutsideListener = (event) => {
-      if (!popover.contains(event.target)) {
-        // User clicked outside, reset to not searching state
-        setSearchTerm('');
-        setAriaHelperText('');
-        setUsers([]);
+    useEffect(() => {
+      if (searchTerm.length < MIN_SEARCH_CHARACTERS) {
+        return;
       }
-    };
-
-    document.addEventListener('click', closeOnClickOutsideListener);
 
-    return () =>
-      document.removeEventListener('click', closeOnClickOutsideListener);
-  }, [searchTerm]);
+      if (cachedSearches[searchTerm]) {
+        setUsers(cachedSearches[searchTerm]);
+        return;
+      }
 
-  useLayoutEffect(() => {
-    const { current: plainTextInput } = plainTextAreaRef;
-    const { current: combobox } = comboboxRef;
+      fetchSuggestions(searchTerm).then(({ result: fetchedUsers }) => {
+        const resultLength = Math.min(
+          fetchedUsers.length,
+          MAX_RESULTS_DISPLAYED,
+        );
```

Comment:
Do we have any documentation for `fetchSuggestions`? 🤔 Maybe I missed it while reading, but might be nice to include it (it took me a minute to figure out that we're taking the lowest value between the results returned from the backend and the max results to display and why we do that)

Reply:
I've added a small inline comment, but it's just a guard since it was a requirement in the RFC to never show more than 6 users in the dropdown. 

I believe there will be some backend changes as part of search 2.0 that will allow us to set the max number of results on the 'search by username' call, but given this component doesn't directly call that endpoint (`fetchSuggestions` is a prop that gets passed in), I think it makes sense to do this check from the UI side too.